module/metrics/rest_api.go

```diff
@@ -94,20 +68,45 @@ func NewRestCollector(cfg metricsProm.Config) module.RestMetrics {
 	return r
 }
 
-// These methods are called automatically by go-http-metrics/middleware
+// ObserveHTTPRequestDuration records the duration of the REST request.
+// This method is called automatically by go-http-metrics/middleware
 func (r *RestCollector) ObserveHTTPRequestDuration(_ context.Context, p httpmetrics.HTTPReqProperties, duration time.Duration) {
-	r.httpRequestDurHistogram.WithLabelValues(p.Service, p.ID, p.Method, p.Code).Observe(duration.Seconds())
+	handler := r.mapURLToRoute(p.ID)
+	r.httpRequestDurHistogram.WithLabelValues(p.Service, handler, p.Method, p.Code).Observe(duration.Seconds())
 }
 
+// ObserveHTTPResponseSize records the response size of the REST request.
+// This method is called automatically by go-http-metrics/middleware
 func (r *RestCollector) ObserveHTTPResponseSize(_ context.Context, p httpmetrics.HTTPReqProperties, sizeBytes int64) {
-	r.httpResponseSizeHistogram.WithLabelValues(p.Service, p.ID, p.Method, p.Code).Observe(float64(sizeBytes))
+	handler := r.mapURLToRoute(p.ID)
+	r.httpResponseSizeHistogram.WithLabelValues(p.Service, handler, p.Method, p.Code).Observe(float64(sizeBytes))
 }
 
+// AddInflightRequests increments and decrements the number of inflight request being processed.
+// This method is called automatically by go-http-metrics/middleware
 func (r *RestCollector) AddInflightRequests(_ context.Context, p httpmetrics.HTTPProperties, quantity int) {
-	r.httpRequestsInflight.WithLabelValues(p.Service, p.ID).Add(float64(quantity))
+	handler := r.mapURLToRoute(p.ID)
+	r.httpRequestsInflight.WithLabelValues(p.Service, handler).Add(float64(quantity))
 }
 
-// New custom method to track all requests made for every REST API request
-func (r *RestCollector) AddTotalRequests(_ context.Context, method string, routeName string) {
-	r.httpRequestsTotal.WithLabelValues(method, routeName).Inc()
+// AddTotalRequests records all REST requests
+// This is a custom method called by the REST handler
+func (r *RestCollector) AddTotalRequests(_ context.Context, method, path string) {
+	handler := r.mapURLToRoute(path)
+	r.httpRequestsTotal.WithLabelValues(method, handler).Inc()
+}
+
+// mapURLToRoute uses the urlToRouteMapper callback to convert a URL to a route name
+// This normalizes the URL, removing dynamic information converting it to a static string
+func (r *RestCollector) mapURLToRoute(url string) string {
+	if r.urlToRouteMapper == nil {
```

Comment:
Looking at the code, this callback should never be set to nil. We should return an error in the constructor if it is, that should indicate a bug.